src/VecSim/containers/data_blocks_container.cpp

```diff
@@ -0,0 +1,52 @@
+#include "data_blocks_container.h"
+
+DataBlocksContainer::DataBlocksContainer(size_t blockSize, size_t elementBytesCount,
+                                         std::shared_ptr<VecSimAllocator> allocator,
+                                         unsigned char _alignment)
+    : VecsimBaseObject(allocator), RawDataContainer(), element_bytes_count(elementBytesCount),
+      len(0), blocks(allocator), block_size(blockSize), alignment(_alignment) {}
+
+DataBlocksContainer::~DataBlocksContainer() = default;
+
+size_t DataBlocksContainer::size() const { return len; }
+
+RawDataContainer_Status DataBlocksContainer::appendElement(const void *element) {
+    if (len % block_size == 0) {
+        blocks.push_back(DataBlock(this->block_size, this->element_bytes_count, this->allocator,
+                                   this->alignment));
+    }
+    blocks.back().addElement(element);
+    len++;
+    return RAW_DATA_CONTAINER_OK;
+}
+
+const char *DataBlocksContainer::getElement(size_t id) const {
+    if (id >= len) {
+        return nullptr;
+    }
+    return blocks.at(id / this->block_size).getElement(id % this->block_size);
+}
+
+RawDataContainer_Status DataBlocksContainer::removeElement(size_t id) {
+    if (id >= len) {
+        return RAW_DATA_CONTAINER_ID_NOT_EXIST;
+    }
+    if (id < len - 1) {
+        return RAW_DATA_CONTAINER_ERR; // only the last element can be removed
```

Comment:
can we add coverage for this? can it ever happen? if not, consider using an assertion